components/style/rule_tree/mod.rs

```diff
@@ -580,13 +586,13 @@ impl StrongRuleNode {
                 let existing =
                     next_sibling_ptr.compare_and_swap(ptr::null_mut(),
                                                       new_ptr,
-                                                      Ordering::SeqCst);
+                                                      Ordering::Acquire);
```

Comment:
> In that case, the relaxed CAS itself wouldn't be visible either,
> so there's no way any other thread would be able to access that pointer, right?

This is the heart of my whole concern here. There are 3 operations that happen here:
(1) The allocator selects an address in memory and returns it as a pointer.
(2) The caller writes various initialization patterns to the region in memory referenced by the pointer.
(3) The caller CASes the pointer into the rule tree, expose it to other threads.

With Relaxed semantics, what prevents another thread observing the ordering (1) (3) (2)? If nothing does, then the thread may see garbage memory.